commons-component-common/src/main/java/org/exoplatform/commons/notification/impl/service/WebNotificationServiceImpl.java

```diff
@@ -69,29 +69,39 @@ public void markAllRead(String userId) throws Exception {
   @Override
   public List<String> get(WebNotificationFilter filter, int offset, int limit) {
     List<String> result = new ArrayList<String>();
-    List<NotificationInfo> gotList = getNotificationInfos(filter, offset, limit);
+    List<NotificationInfo> gotList = getNotificationInfos(filter, offset, 20);
     NotificationContext ctx = NotificationContextImpl.cloneInstance();
     ctx.append(POPUP_OVER, filter.isOnPopover());
     AbstractChannel channel = ctx.getChannelManager().getChannel(ChannelKey.key(WebChannel.ID));
-    //
-    for (NotificationInfo notification : gotList) {
-      AbstractTemplateBuilder builder = channel.getTemplateBuilder(notification.getKey());
-      MessageInfo msg = null;
-      try {
-        msg = builder.buildMessage(ctx.setNotificationInfo(notification));
-      } catch (Exception e) {
-        LOG.error("Error while building message for notification with id = " + notification.getId(), e);
+    while (true) {
+      for (NotificationInfo notification : gotList) {
+        AbstractTemplateBuilder builder = channel.getTemplateBuilder(notification.getKey());
+        MessageInfo msg = null;
+        try {
+          msg = builder.buildMessage(ctx.setNotificationInfo(notification));
+        } catch (Exception e) {
+          LOG.error("Error while building message for notification with id = " + notification.getId(), e);
+        }
+        if (msg != null && msg.getBody() != null && !msg.getBody().isEmpty()) {
+          result.add(msg.getBody());
+        }
+        // if have any exception when template transformation
+        // ignore to display the notification
+        if (ctx.isFailed()) {
+          LOG.warn(ctx.getException().getMessage(), ctx.getException());
+        }
+        if (result.size() == limit){
```

Comment:
This condition supposes that we have more notifications in database than the limit. Which is not corretc. That leads to an infinite loop.

Reply:
this condition is used to avoid to assigne other objects to result when its size is equal to Liimit when there are other notif in gotList